examples/dao/assets/dao-app-approval.py

```diff
@@ -0,0 +1,471 @@
+    # scratch vars to save result() & scratch_proposal_active()
+    scratchvar_result = ScratchVar(TealType.uint64)
+    scratchvar_proposal_active = ScratchVar(TealType.uint64)
+
+    # scratch vars to store proposal config
+    scratchvar_voting_start = ScratchVar(TealType.uint64)
+    scratchvar_voting_end = ScratchVar(TealType.uint64)
+    scratchvar_execute_before = ScratchVar(TealType.uint64)
+    scratchvar_proposal_type = ScratchVar(TealType.uint64)
+
+    # Fetch key(s) from Txn.Accounts[1] (passed as proposalLsig during execute), for current app
+    # these are used throughout the application
+    # NOTE: during add_proposal, proposalLsig is the sender
+    voting_start = App.localGet(Int(1), Bytes("voting_start"))
+    voting_end = App.localGet(Int(1), Bytes("voting_end"))
+    proposal_id = App.localGet(Int(1), Bytes("id"))
+    yes_count = App.localGet(Int(1), Bytes("yes"))
+    no_count = App.localGet(Int(1), Bytes("no"))
+    execute_before = App.localGet(Int(1), Bytes("execute_before"))
+    executed = App.localGet(Int(1), Bytes("executed"))
+    proposal_type = App.localGet(Int(1), Bytes("type"))
+    proposal_from = App.localGet(Int(1), Bytes("from"))
+    asa_id = App.localGet(Int(1), Bytes("asa_id"))
+    recipient = App.localGet(Int(1), Bytes("recipient"))
+    amount = App.localGet(Int(1), Bytes("amount"))
+
+    # Computes result of proposal. Saves result in scratch. Args:
+    # * idx - index of account where proposal_lsig.address() is passed
+    # NOTE: idx == Int(0) means proposalLsig is Txn.sender()
+    def compute_result(idx: Int):
+        return Seq([
+            Cond(
+                # 4 if proposal expired (now > proposal.execute_before())
+                [Global.latest_timestamp() > App.localGet(idx, Bytes("execute_before")), scratchvar_result.store(Int(4))],
+                # 3 if voting is still in progress (now <= voting_end)
+                [Global.latest_timestamp() <= App.localGet(idx, Bytes("voting_end")), scratchvar_result.store(Int(3))],
+                # 1 if voting is over and proposal.yes >= min_support and proposal.yes > proposal.no
+                [
+                    And(
+                        Global.latest_timestamp() > App.localGet(idx, Bytes("voting_end")),
+                        App.localGet(idx, Bytes("yes")) >= App.globalGet(min_support),
+                        App.localGet(idx, Bytes("yes")) > App.localGet(idx, Bytes("no"))
+                    ) == Int(1),
+                    scratchvar_result.store(Int(1))
+                ],
+            ),
+            # if proposal is not expired, not in progess, and not passed, then reject (set result == Int(2))
+            If(
+                And(
+                    scratchvar_result.load() != Int(4), # expired
+                    scratchvar_result.load() != Int(3), # in progess
+                    scratchvar_result.load() != Int(1), # passed
+                ),
+                scratchvar_result.store(Int(2))
+            )
```

Comment:
instead of having the `If`, maybe we can add put into conditions (at the end):
```
[true, scratchvar_result.store(Int(1))]
```

Reply:
I was putting it into conditions first, but an error is thrown that "we cannot load from scratch before store". Hence i put it after the conditions.

Reply:
strange, in `scratchvar_result.store` we shouldn't load a value - it only saves the value.

Reply:
yeah, it's a pyTEAL error (when compiling pyTEAL -> TEAL).